pkg/controllers/monitoring-stack/components.go

```diff
@@ -416,6 +447,30 @@ func newRoleBinding(ms *stack.MonitoringStack, rbacResourceName string) *rbacv1.
 	return roleBinding
 }
 
+func newService(name string, ms *stack.MonitoringStack, instanceSelectorKey string, instanceSelectorValue string) *corev1.Service {
```

Comment:
shouldn't the creation of these resources be validated in a test?

Reply:
Yes, this is a good point. I would just try to validate correctness by issuing queries against the service instead of running assertions on the service object fields. The reason for this is because the former is a proper behavior test, and the latter feels a bit like testing an implementation detail.